docs/build/voting-dapp-tutorial.md

```diff
@@ -0,0 +1,1245 @@
+
+  (defun vote (key)
+    "Submit a new vote"
+    (format "Voted {}!" [key]))
+)
+
+(vote "A")
+```
+
+This snippet defines a module "election" that holds a function "vote" that takes a parameter and displays a formatted string that includes the parameter.
+
+:::note
+Module names and keyset definitions are required to be unique. We will mention this again when we get to deploy our contract to testnet.
+:::
+
+Before we move forward, let's quickly test our code. Copy the code below in the `election.repl` file:
+
+```clojure
+;; begin a transaction
+(begin-tx)
+;; set transaction signature key to my-key
+(env-keys ["my-key"])
+;; Add the election-admin-keyset to environment data
+(env-data {
+  'election-admin-keyset: { "keys": ["admin-key"], "pred": "keys-all" }  
+})
+;; load election.pact into the REPL
+(load "election.pact")
+;; commit the transaction
+(commit-tx)
+```
+
+:::info
+`begin-tx`, `env-keys`, `env-data` are "repl-only" functions, they are automatically loaded in the REPL to help simulate blockchain environment but they are not available for blockchain-based execution. You can find the complete "REPL-only" functions list [here](https://pact-language.readthedocs.io/en/latest/pact-functions.html?highlight=repl%20only#repl-only-functions).
+:::
+
+To run the test type the following command in your terminal:
+```clojure
+$ pact
+pact> (load "election.repl")
+```
+And you should see an output similar to the one below (the hash will be different):
+```clojure
+"Loading election.repl..."
+"Begin Tx 0"
+"Setting transaction keys"
+"Setting transaction data"
+"Loading election.pact..."
+"Keyset defined"
+"Loaded module election, hash jOBJiSEH5HVsAmLNSdHq_D3Kl6TkFpCJcEWPtJD86Gc"
+"Voted A!"
+"Commit Tx 0"
+```
+
+Let's continue by adding a bit more logic. Copy this snippet just below the module declaration:
+
+```clojure
+  (defschema candidates-schema
+    "Candidates table schema"
+    name:string
+    votes:integer)
+
+  (defschema votes-schema
+    "Votes table schema"
+    cid:string
+  )
+
+  (deftable votes:{votes-schema})
+
+  (deftable candidates:{candidates-schema})
+  ```
+In the snippet we've got several definitions:
+* `defschema candidates-schema` -> defines the schema for our **candidates** table where we will keep track of the candidates that users can vote for
+* `defschema votes-schema` -> defines the schema for our **votes** table where we will store each vote
+* `deftable votes:{votes-schema}` -> defines the `votes` table that will use the schema defined above
+* `deftable candidates:{candidates-schema}` -> defines the `candidates` table that will use the schema defined above
+
+:::note
+Pact implements a key-row model which means a row is accessed by a single key. It is our responsibility as developers to design the schema in a way that we can retrieve the information that we need using a single row query. Multiple row queries are expensive which we will see in an example a bit later.
```

Comment:
Make a note that the key is always a string. Maybe also note that "key" here is not the same as a "cryptographic key" that we use for signing? I wonder if that might get confusing

Reply:
Hmm, it should not be for anyone that worked with key-value dbs before. I left note though just in case.